File: frontend/src/types/session.ts
export interface Session {
  id: string
  project_id: string
  title: string
  message_count: number
  created_at: string
  updated_at: string
}

export interface Message {
  id: string
  session_id: string
  role: 'user' | 'assistant'
  content: string
  sources?: { table: string; column?: string; description?: string }[]
  created_at: string
}

export interface AskRequest {
  question: string
  session_id?: string
}

export interface AskSyncResponse {
  answer: string
  sources?: { table: string; column?: string; description?: string }[]
  session_id: string
}

export interface SessionCreate {
  project_id: string
  title?: string
}

File: backend/app/schemas/session.py
from datetime import datetime

from pydantic import BaseModel


class SessionCreate(BaseModel):
    project_id: str
    title: str | None = None


class SessionResponse(BaseModel):
    id: str
    project_id: str
    title: str | None = None
    created_at: datetime
    updated_at: datetime
    message_count: int = 0


class SessionListResponse(BaseModel):
    items: list[SessionResponse]
    total: int


class MessageResponse(BaseModel):
    id: str
    session_id: str
    role: str
    content: str
    sources: list | None = None
    created_at: datetime


class MessageListResponse(BaseModel):
    items: list[MessageResponse]
    total: int
